src/app/request/pay/page.tsx

```diff
@@ -1,28 +1,73 @@
 import { Metadata } from 'next'
 import Layout from '@/components/Global/Layout'
-import * as components from '@/components'
+import { PayRequestLink } from '@/components'
+import { headers } from 'next/headers'
+import { PreviewType } from '@/components/Global/ImageGeneration/LinkPreview'
+import { IRequestLinkData } from '@/components/Request/Pay/Pay.consts'
+import { formatAmount } from '@/utils'
 
-export const metadata: Metadata = {
-    // TODO: make metadata dynamic and change title based on if payment completed or not
-    // see claim/page.tsx
-    title: 'Peanut Protocol - Payment Request',
-    description: 'Text Tokens',
-    metadataBase: new URL('https://peanut.to'),
-    icons: {
-        icon: '/logo-favicon.png',
-    },
-    openGraph: {
-        images: [
-            {
-                url: '/metadata-img.png',
+export const dynamic = 'force-dynamic'
+
+type Props = {
+    params: { id: string }
+    searchParams: { [key: string]: string | string[] | undefined }
+}
+
+function getPreviewUrl(host: string, data: IRequestLinkData): string {
+    const url = new URL('/api/preview-image', host)
+
+    const params = new URLSearchParams({
+        amount: data.tokenAmount,
+        chainId: data.chainId,
+        tokenAddress: data.tokenAddress,
+        tokenSymbol: data.tokenSymbol ?? '',
+        address: data.recipientAddress,
+        previewType: PreviewType.REQUEST,
+    })
+
+    url.search = params.toString()
+    return url.toString()
+}
+
+export async function generateMetadata({ searchParams }: Props): Promise<Metadata> {
+    let title = 'Request Payment'
+    let previewUrl = '/metadata-img.jpg'
+    let host = headers().get('host') || 'peanut.to'
+    host = `${process.env.NODE_ENV === 'development' ? 'http://' : 'https://'}${host}`
+    try {
+        const linkRes = await fetch(`${host}/api/proxy/get/request-links/${searchParams.id}`, {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json',
             },
-        ],
-    },
+        })
+        if (linkRes.ok) {
+            const linkDetails: IRequestLinkData = await linkRes.json()
+            title = `${linkDetails.recipientAddress} is requesting ${formatAmount(Number(linkDetails.tokenAmount))} ${linkDetails.tokenSymbol}`
```

Comment:
Imo we should do as few direct calls to API as possible and try to dogfood our SDK if appropriate. In this case we should use `await peanut.getRequestLinkDetails({`


Re: https://www.notion.so/peanutprotocol/Request-data-is-not-stored-properly-in-every-case-eg-tokenSymbol-c6f6fa0909f84d4b85263c92a3b98faf
Also, current implementation seems to not resolve tokenSymbol. Worst case we can fetch that realtime from onchain, we have an sdk function for that, erc20 contracts have that info online

Reply:
Sounds good, does it make sense to modify the sdk getRequestLinkDetails to also accept a uuid directly? It seems pointless to create an url here just for that, thats the main reason why i called directly the API

About the token symbol, in general (and specially in the metadata call) I prefer if we can assume some things, for example: a request link in the db to have tokenSymbol and if that's not the case right now do all the stuff when creating the link (fetching from onchain for example)